tests/test_optuna_transformers.py

```diff
@@ -0,0 +1,168 @@
+# Copyright (c) 2021 Timothy Wolff-Piggott
+# This software is distributed under the terms of the MIT license
+# which is available at https://opensource.org/licenses/MIT
+
+import tempfile
+from pathlib import Path
+from typing import Dict
+
+import pytest
+import sklearn.model_selection
+import torch
+from mlflow.tracking import MlflowClient
+from transformers import (
+    ElectraConfig,
+    ElectraForSequenceClassification,
+    ElectraTokenizer,
+    EvalPrediction,
+    Trainer,
+    TrainingArguments,
+)
+
+import optuna
+from hpoflow.optuna_mlflow import OptunaMLflow
+from hpoflow.optuna_transformers import OMLflowCallback
+
+# global parameters for training
+MAX_STEPS = 4
+# an eval step is only logged if it coincides with a logging step
+# logging steps should divide eval steps otherwise some evals are lost
+LOGGING_STEPS = EVAL_STEPS = 2
+assert (
+    EVAL_STEPS % LOGGING_STEPS == 0
+), f"Logging steps {LOGGING_STEPS} should divide eval steps {EVAL_STEPS}"
+N_EVAL_STEPS = MAX_STEPS // EVAL_STEPS
+LAST_EVAL_STEP = N_EVAL_STEPS * EVAL_STEPS
+N_TRIALS = 2
+
+
+@pytest.fixture(scope="module")
+def config():
+    return ElectraConfig()
+
+
+@pytest.fixture(scope="module")
+def model(config):
+    return ElectraForSequenceClassification(config)
+
+
+@pytest.fixture(scope="module")
+def output_path():
+    tempdir = tempfile.mkdtemp()
+    return Path(tempdir)
+
+
+@pytest.fixture(scope="module")
+def tracking_file_name(output_path):
+    return f"file:{output_path.as_posix()}"
+
+
+def test_omlflow_callback(config, output_path, model, tracking_file_name):
```

Comment:
Should we use the tmpdir-fixture or tmp-path-fixture from pytest here instead of `tracking_file_name`?
https://docs.pytest.org/en/6.2.x/tmpdir.html

Reply:
Thanks, I've updated to use `tmp_path_factory` to create a single module-scoped tmpdir fixture. `tmp_path` is only function-scoped I think

https://pytest.org/en/latest/how-to/tmpdir.html#tmpdir

Reply:
This allows us to access the same file in both the training and test functions

Reply:
Ok.